ls1intum/Artemis
# `Development`: Move assessment due date check into ExerciseDateService

### Checklist
#### General
#### Server

### Motivation and Context
Before the assessment due date, the student should under no circumstances see the manual tutor feedback. However, not every endpoint is currently checking and enforcing this rule.

### Description
This Refactoring is the first step to consistently check and remove scores and feedback on the server side if the student is not yet allowed to see it.

This PR adds the after assessment due date checks to the `ExerciseDateService` and uses these methods instead of always directly comparing the date with `now()`. This also makes the code more readable, since you now have a descriptive method name in these places and don't need to think yourself about the condition and what is meant by it.

### Steps for Testing
This PR requeires cafeull testsing and code review!

Please go through the code and check that the logic ist still correct (and was not unintentionally inverted).

1. Create exercises (every type) and participate in them as a student.
2. Grade the submissions as a tutor.
3. Check different places on Artemis (e.g. exercise detail, exercise overview, statistics, front page, etc.). Check that the student is never able to see the result generated by the tutor (before the assessment due date).
4. After the assessment due date, check the the result is now shown.

#### Exam Mode Testing
same as above, but with exam exercises. No results should be visible before the "publish exam results" date is reached.

### Review Progress

#### Code Review
#### Manual Tests
#### Exam Mode Test

### Test Coverage
added tests for ExerciseDateService (coverage unchanged since it's also covered by other integration tests)
Adjusted existing tests to check for filtering.
